haystack/components/embedders/azure_document_embedder.py

```diff
@@ -59,6 +59,7 @@ def __init__(  # noqa: PLR0913 (too-many-arguments) # pylint: disable=too-many-p
         default_headers: Optional[Dict[str, str]] = None,
         azure_ad_token_provider: Optional[AzureADTokenProvider] = None,
         http_client_kwargs: Optional[Dict[str, Any]] = None,
+        raise_on_failure: bool = False,
```

Comment:
I guess we don't want to introduce a potentially breaking change here, right ? 

I think the many other components rather fail then skip errors during execution. Does it make sense to set the default behaviour to True to unify the behaviour ?